### DISCUSSION TITLE
Fitting Progress Freezing at 1000 iterations

### DISCUSSION THREAD

**Discussion by a-wright20** [refnx/refnx#646] [Q&A]
Hello, 

I'm new to using Refnx and I recently came upon a problem where the Motofit gui will freeze when it hits 1000 iterations. I am batch-fitting XR profiles of monolayers on liquid surfaces, and I believe this is taking so many iterations because I have several linked parameters across my system. Regardless, the Fitting Progress window freezes (and gives a 'Not Responding') and I was wondering if there was some way to fix this.

**Comment by andyfaff** [comment]
@wizboi1357, it's going to be hard to diagnose or fix the problem without fitting under the same conditions. Would it be possible for you to send me the saved experiment file ("File --> Save Experiment As"), and a list of instructions with what to do in the GUI, so I can try and repeat what you see?

**Reply by a-wright20** [reply]
[PFDA_DI_multifitting_w_gui.zip](https://github.com/refnx/refnx/files/9749531/PFDA_DI_multifitting_w_gui.zip)

I believe I attached the file. Thank you so much for your help! I believe I have all the fitting parameters selected with the upper and lower bounds filled out. I fit all three data sets by moving them into the box at the lower right. The head SLD parameter is linked as equivalent across all three data sets; and within each set, the roughness is linked for the head and tail.

**Comment by andyfaff** [comment]
Some first impressions:

- probably don't fit dq - resolution. XRR measurements are typically high resolution, so I normally set this value to 0 and don't fit it. XRR measurements typically aren't constant dq/q resolution anyway.
- I wouldn't fit a Q-offset without a well defined critical edge.

**Comment by andyfaff** [comment]
I then changed the bkg fitting parameter limits. Make the lower limit 1e-10, upper limit 1e-8 for all three. Limits of [0, 2] don't make sense there.
When I did all three of these points the fit occurred without any issues.

**Reply by a-wright20** [reply]
Thank you for the advice on the parameters. 

However, even when I change those it still stops fitting at 1000 iterations. I wonder if it is a problem with the python files I have installed. Thank you for your help

**Comment by andyfaff** [comment]
Try pressing the green fit button with the attached file, it works for me, and doesn't hang.
[PFDA_DI_multifitting_w_gui.mtft.zip](https://github.com/refnx/refnx/files/9750599/PFDA_DI_multifitting_w_gui.mtft.zip)

- I removed `dq-resolution` from the list of fitted parameters, setting dq =0.
- I tightened the bounds for the bkg parameters
- I removed all the `q - offset` parameters.

Can you tell me the contents of the box in the menu item "Help --> Python Packages". This will list the package versions you are using.

**Reply by a-wright20** [reply]
Sure, here are the packages I have installed:
refnx == 0.1.29
scipy == 1.9.1
numpy == 1.23.3
periodictable == 1.5.2
matplotlib == 3.6.0

Your file ran fine on my system. I guess those changes really make a difference. Thank you for your help.
